File: ts/src/tools/DragAndDropTool.ts
import { BaseTool } from './BaseTool';
import { BaseDriver } from '../drivers/BaseDriver';

export class DragAndDropTool extends BaseTool {
  static description = "Drag one element onto another and drop it. Don't combine with HoverTool.";

  fromId: number;
  toId: number;

  constructor(args: { from_id: number; to_id: number }) {
    super();
    this.fromId = args.from_id;
    this.toId = args.to_id;
  }

  invoke(driver: BaseDriver): void {
    driver.dragAndDrop(this.fromId, this.toId);
  }
}

File: packages/java/src/main/java/ai/alumnium/tool/DragAndDropTool.java
package ai.alumnium.tool;

import ai.alumnium.driver.BaseDriver;
import ai.alumnium.tool.annotation.ToolDescription;
import ai.alumnium.tool.annotation.ToolField;

@ToolDescription("Drag one element onto another and drop it. Don't combine with HoverTool.")
public record DragAndDropTool(
    @ToolField(description = "Identifier (ID) of element to drag") int fromId,
    @ToolField(description = "Identifier (ID) of element to drop onto") int toId)
    implements BaseTool {
  @Override
  public void invoke(BaseDriver driver) {
    driver.dragAndDrop(fromId, toId);
  }
}
